bee-message/tests/parents.rs

```diff
@@ -93,37 +98,39 @@ fn pack_unpack_invalid_more_than_max() {
     ];
 
     assert!(matches!(
-        Parents::unpack(&mut bytes.as_slice()),
-        Err(Error::InvalidParentsCount(9))
+        Parents::unpack_verified(&mut bytes.as_slice()),
+        Err(UnpackError::Packable(Error::InvalidParentCount(
+            TryIntoBoundedU8Error::Invalid(9)
+        )))
     ));
 }
 
 #[test]
 fn unpack_invalid_not_sorted() {
     let mut inner = rand_message_ids(8);
     inner.reverse();
+    let inner = VecPrefix::<_, u8>::try_from(inner).unwrap();
 
-    // Remove 8 byte vector length field and replace with 1 byte, to represent message parents.
-    let mut packed = (8u8).pack_new();
-    let mut packed_messages = inner.pack_new().split_at(core::mem::size_of::<u64>()).1.to_vec();
-    packed.append(&mut packed_messages);
-
-    let parents = Parents::unpack(&mut packed.as_slice());
+    let packed = inner.pack_to_vec();
+    let parents = Parents::unpack_verified(&mut packed.as_slice());
 
-    assert!(matches!(parents, Err(Error::ParentsNotUniqueSorted)));
+    assert!(matches!(
+        parents,
+        Err(UnpackError::Packable(Error::ParentsNotUniqueSorted))
```

Comment:
I'm a bit confused by this test, what makes it not unique sorted with guarantee ?

Reply:
I just ported it but I guess it has to do with the `reverse()` call.

Reply:
I'm asking because I saw you removed some line and I assume they were dealing with this. I'm having a hard time understanding the test...

Reply:
Why would reversing something random guarantee that it's unique sorted ?

Reply:
Ahhhh, don't mind me `rand_message_ids` actually sorts